lib/address.ml

```diff
@@ -0,0 +1,27 @@
+open Bin_prot.Std
+
+type t = {
+  address : string;
+  port : int;
+}
+[@@deriving bin_io, eq]
+
+let address_of address =
```

Comment:
Hmm, since we are providing these functions and the address type is not mutable, what if we just expose the fields of the type in the mli? Seems like no harm would be done if we did